meson_cpu/x86/meson.build

```diff
@@ -214,6 +214,10 @@ if compiler_id == 'msvc'
   FMA3.update(args: {'val': '/arch:AVX2', 'match': clear_arch})
   AVX2.update(args: {'val': '/arch:AVX2', 'match': clear_arch})
   AVX512_SKX.update(args: {'val': '/arch:AVX512', 'match': clear_arch})
+  
+  # Add floating-point contract flag for 32-bit builds
+  # Fixes transcendental function accuracy on Windows Server 2022
+  add_project_arguments('/fp:contract', language: ['c', 'cpp'])
```

Comment:
This should depend on the existence of FMA. Note that `add_project_arguments` is not used elsewhere in cpu specific contexts.